{"expand":"renderedFields,names,schema,operations,editmeta,changelog,versionedRepresentations","id":"12384540","self":"https://issues.apache.org/jira/rest/api/2/issue/12384540","key":"OPENJPA-466","fields":{"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313301","id":"12313301","description":"","name":"1.0.4","archived":false,"released":true,"releaseDate":"2010-09-03"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313681","id":"12313681","description":"","name":"1.2.2","archived":false,"released":true,"releaseDate":"2010-01-18"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313326","id":"12313326","description":"","name":"1.3.0","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12314148","id":"12314148","description":"Early Access 3","name":"2.0.0-M3","archived":false,"released":true,"releaseDate":"2009-10-12"}],"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12312322":null,"customfield_12312323":null,"customfield_12312320":null,"customfield_12310420":"41868","customfield_12312321":null,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312720":null,"customfield_12312325":null,"lastViewed":null,"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/1","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/blocker.svg","name":"Blocker","id":"1"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12313422":"false","customfield_12312331":null,"customfield_12312332":null,"aggregatetimeoriginalestimate":null,"timeestimate":null,"customfield_12312330":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12312341","id":"12312341","description":"","name":"1.0.0","archived":false,"released":true,"releaseDate":"2007-08-28"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12312687","id":"12312687","description":"","name":"1.0.1","archived":false,"released":true,"releaseDate":"2007-11-09"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12312344","id":"12312344","description":"","name":"1.1.0","archived":false,"released":true,"releaseDate":"2008-05-22"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313102","id":"12313102","description":"","name":"1.2.0","archived":false,"released":true,"releaseDate":"2008-08-12"}],"customfield_12311120":null,"customfield_12313826":null,"customfield_12312339":null,"issuelinks":[{"id":"12327606","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12327606","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12439611","key":"OPENJPA-1372","self":"https://issues.apache.org/jira/rest/api/2/issue/12439611","fields":{"summary":"Generating identifiers by using sequence table may fail during sequence table initialization","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133}}}}],"customfield_12313825":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"customfield_12312337":null,"customfield_12313823":null,"customfield_12312338":null,"customfield_12311920":null,"customfield_12313822":null,"customfield_12312335":null,"customfield_12313821":null,"customfield_12312336":null,"customfield_12313820":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"components":[],"customfield_12312026":null,"customfield_12312023":null,"customfield_12312024":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12310921":null,"customfield_12310920":"201583","customfield_12312823":null,"creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=frankca","name":"frankca","key":"frankca","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Frank Le","active":true,"timeZone":"Etc/UTC"},"subtasks":[],"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=frankca","name":"frankca","key":"frankca","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Frank Le","active":true,"timeZone":"Etc/UTC"},"aggregateprogress":{"progress":0,"total":0},"customfield_12313520":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"customfield_12313924":null,"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/OPENJPA-466/votes","votes":5,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12313920":null,"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133},"timespent":null,"customfield_12314020":"{summaryBean=com.atlassian.jira.plugin.devstatus.rest.SummaryBean@3f2fb580[summary={pullrequest=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@1ca2e1b4[overall=PullRequestOverallBean{stateCount=0, state='OPEN', details=PullRequestOverallDetails{openCount=0, mergedCount=0, declinedCount=0}},byInstanceType={}], build=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@7b2d2c3b[overall=com.atlassian.jira.plugin.devstatus.summary.beans.BuildOverallBean@7009ddab[failedBuildCount=0,successfulBuildCount=0,unknownBuildCount=0,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], review=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@3286ea6d[overall=com.atlassian.jira.plugin.devstatus.summary.beans.ReviewsOverallBean@6b5a7c1b[stateCount=0,state=<null>,dueDate=<null>,overDue=false,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], deployment-environment=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@7984d9bf[overall=com.atlassian.jira.plugin.devstatus.summary.beans.DeploymentOverallBean@475eb7a1[topEnvironments=[],showProjects=false,successfulCount=0,count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], repository=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@420ee8a5[overall=com.atlassian.jira.plugin.devstatus.summary.beans.CommitOverallBean@7a8cee5b[count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}], branch=com.atlassian.jira.plugin.devstatus.rest.SummaryItemBean@259a9994[overall=com.atlassian.jira.plugin.devstatus.summary.beans.BranchOverallBean@17cbd139[count=0,lastUpdated=<null>,lastUpdatedTimestamp=<null>],byInstanceType={}]},errors=[],configErrors=[]], devSummaryJson={\"cachedValue\":{\"errors\":[],\"configErrors\":[],\"summary\":{\"pullrequest\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"stateCount\":0,\"state\":\"OPEN\",\"details\":{\"openCount\":0,\"mergedCount\":0,\"declinedCount\":0,\"total\":0},\"open\":true},\"byInstanceType\":{}},\"build\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"failedBuildCount\":0,\"successfulBuildCount\":0,\"unknownBuildCount\":0},\"byInstanceType\":{}},\"review\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"stateCount\":0,\"state\":null,\"dueDate\":null,\"overDue\":false,\"completed\":false},\"byInstanceType\":{}},\"deployment-environment\":{\"overall\":{\"count\":0,\"lastUpdated\":null,\"topEnvironments\":[],\"showProjects\":false,\"successfulCount\":0},\"byInstanceType\":{}},\"repository\":{\"overall\":{\"count\":0,\"lastUpdated\":null},\"byInstanceType\":{}},\"branch\":{\"overall\":{\"count\":0,\"lastUpdated\":null},\"byInstanceType\":{}}}},\"isStale\":false}}","customfield_12314141":null,"customfield_12314140":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/12310351","id":"12310351","key":"OPENJPA","name":"OpenJPA","projectTypeKey":"software","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=12310351&avatarId=10043","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310351&avatarId=10043","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310351&avatarId=10043","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310351&avatarId=10043"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10252","id":"10252","description":"","name":"OpenJPA"}},"aggregatetimespent":null,"customfield_12312520":null,"customfield_12314422":null,"customfield_12314421":null,"customfield_12314146":null,"customfield_12314420":null,"customfield_12314145":null,"customfield_12314144":null,"customfield_12314143":null,"resolutiondate":"2009-10-31T21:13:01.371+0000","workratio":-1,"customfield_12312923":null,"customfield_12312920":null,"customfield_12312921":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/OPENJPA-466/watchers","watchCount":4,"isWatching":false},"created":"2007-12-13T02:40:28.973+0000","updated":"2010-03-09T18:31:00.950+0000","timeoriginalestimate":null,"description":"Here's how I annotate the ID:\n    @Id\n    @SequenceGenerator(name = \"FooSeq\", sequenceName = \"seq_foo\", allocationSize = 20)\n    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = \"FooSeq\")\n    private Long id;\n\nHere's how I create the (Oracle) sequence:\nCREATE SEQUENCE seq_foo START WITH 1 INCREMENT BY 1;\n\nI get a primary key unique constraint violated in a multithreaded app i.e. it doesn't happen in single-threaded!\n\nYou can simply reproduce this error by either create blocking queue or blocking thread pool say size 5 to insert 10000+ object.\n","customfield_10010":null,"timetracking":{},"customfield_12314523":null,"customfield_12314127":null,"customfield_12314522":null,"customfield_12314126":null,"customfield_12314521":null,"customfield_12314125":null,"customfield_12314520":null,"customfield_12314124":null,"attachment":[{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12422577","id":"12422577","filename":"GeneratedIdObject.java","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-10-19T17:15:24.446+0000","size":513,"mimeType":"text/x-java-source","content":"https://issues.apache.org/jira/secure/attachment/12422577/GeneratedIdObject.java"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12406496","id":"12406496","filename":"OPENJPA-466.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-04-27T08:21:33.474+0000","size":1524,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12406496/OPENJPA-466.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12400305","id":"12400305","filename":"OPENJPA-466.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-17T02:40:48.607+0000","size":29757,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12400305/OPENJPA-466.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12417527","id":"12417527","filename":"OPENJPA-466-1.0.x.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bjreed","name":"bjreed","key":"bjreed","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Barrett Jay Reed","active":true,"timeZone":"Etc/UTC"},"created":"2009-08-24T21:08:48.918+0000","size":30589,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12417527/OPENJPA-466-1.0.x.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12417528","id":"12417528","filename":"OPENJPA-466-1.2.x.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bjreed","name":"bjreed","key":"bjreed","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Barrett Jay Reed","active":true,"timeZone":"Etc/UTC"},"created":"2009-08-24T21:08:48.954+0000","size":30591,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12417528/OPENJPA-466-1.2.x.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12413238","id":"12413238","filename":"OPENJPA-466-SYNCRONIZED.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-07-12T15:14:55.526+0000","size":694,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12413238/OPENJPA-466-SYNCRONIZED.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12422576","id":"12422576","filename":"TestSequenceGenerationOnMT.java","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-10-19T17:15:24.427+0000","size":1982,"mimeType":"text/x-java-source","content":"https://issues.apache.org/jira/secure/attachment/12422576/TestSequenceGenerationOnMT.java"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12408717","id":"12408717","filename":"volatile.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-21T16:58:25.003+0000","size":570,"mimeType":"text/x-patch","content":"https://issues.apache.org/jira/secure/attachment/12408717/volatile.patch"}],"customfield_12312340":null,"customfield_12314123":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12314122":null,"customfield_12314121":null,"customfield_12310041":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10042","value":"Patch Available","id":"10042","disabled":false}],"customfield_12314120":null,"customfield_12314129":null,"customfield_12314524":null,"customfield_12314128":null,"summary":"Primary key constraint violated using (Oracle) sequence to generate ID in multithreaded app","customfield_12314130":null,"customfield_12310291":null,"customfield_12310290":null,"customfield_12314138":null,"customfield_12314137":null,"environment":"OpenJPA 1.0.0 (also tried 1.0.1 and 1.1.0-SNAPSHOT)\nOracle XE 10g (JDBC driver 10.2.0.3.0)\nWindows XP Pro","customfield_12314136":null,"customfield_12314135":null,"customfield_12311020":null,"customfield_12314134":null,"duedate":null,"customfield_12314132":null,"customfield_12314131":null,"comment":{"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12572281","id":"12572281","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=gmarkham","name":"gmarkham","key":"gmarkham","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Gil Markham","active":true,"timeZone":"Etc/UTC"},"body":"I've seen the same behavior using postgresql 8.2/8.3 and sequences and openjpa 1.0.1 and 1.0.2.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=gmarkham","name":"gmarkham","key":"gmarkham","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Gil Markham","active":true,"timeZone":"Etc/UTC"},"created":"2008-02-25T22:33:00.086+0000","updated":"2008-02-25T22:33:00.086+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12669924","id":"12669924","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jpullen","name":"jpullen","key":"jpullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Joe Pullen","active":true,"timeZone":"Etc/UTC"},"body":"is this issue resolved ? its pretty old and I am experiencing a similar error with version 1.2.0 and the priority is Blocker ?","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jpullen","name":"jpullen","key":"jpullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Joe Pullen","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-03T11:15:58.776+0000","updated":"2009-02-03T11:15:58.776+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12670605","id":"12670605","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Joe, I'm trying to replicate the failure so that I can fix this problem. However, I'm still unable to reproduce it. Are you only seeing it fail in a multi-threaded application ?? And if so, are both threads connected to the same database and/or using the same sequence ?? Thanks for any additional information. ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-05T02:20:47.943+0000","updated":"2009-02-05T02:20:47.943+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12671815","id":"12671815","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jpullen","name":"jpullen","key":"jpullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Joe Pullen","active":true,"timeZone":"Etc/UTC"},"body":"Hi Tim,\n\nDefinately get the issue in a WebLogic multithreaded enviroment with 11g (RAC). The version of  OpenJPA used is 1.2.0.\nThe issues comes with more load (25-100 TPS). The application has about 25 entities all using seqs on the primary key.\nThe error is always on the flush. The seq cache is set to 100.\n\nThe test cases are not mutilthreaded in our app, but maybe I try to get a working, or is that non working ;-) test.\nDo you have ideas where the problem lies ?","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jpullen","name":"jpullen","key":"jpullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Joe Pullen","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-09T11:42:50.041+0000","updated":"2009-02-09T11:42:50.041+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672404","id":"12672404","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Joe, I think I'm finally able to replicate this error !! I haven't run it yet on Oracle, but I'm consistently able to get this failure using postgresql. Does this look similar to the error you're getting in Oracle ?? \n\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:742775 nonfatal general error> org.apache.openjpa.persistence.PersistenceException: ERROR: duplicate key value violates unique constraint \"entity_employee_pkey\" {prepstmnt 17832855 INSERT INTO ENTITY_EMPLOYEE (id, firstName, lastName, salary) VALUES (?, ?, ?, ?) [params=(int) 273350, (String) First_name_1628, (String) Last_name_1628, (float) 1628.0]} [code=0, state=23505]\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-10T21:22:04.493+0000","updated":"2009-02-10T21:22:04.493+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672482","id":"12672482","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"Is openjpa.Multithreaded configuration property is set to 'true'? ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-02-11T02:25:47.099+0000","updated":"2009-02-11T02:25:47.099+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672505","id":"12672505","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Pinaki, in my postgresql testcase I'm setting the openjpa.Multithreaded property in the setup() mehtod as below. Does this look correct ??\n\n    public void setUp() {\n        setUp(EntityPerson.class, \n                EntityEmployee.class, \n        \tCLEAR_TABLES, \"openjpa.Multithreaded\", \"true\");\n    }\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-11T04:39:03.899+0000","updated":"2009-02-11T04:39:03.899+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672606","id":"12672606","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"body":"Hi Tim,\n\nLooks similar to mine from oracle.\n\n]]></stack><cause><stack><![CDATA[      <openjpa-1.2.0-r422266:683325 nonfatal general error> org.apache.openjpa.persistence.PersistenceException: ORA-00001: unique constraint (P\nOR.DSTA_PK) violated\n {prepstmnt 97 INSERT INTO DIA_STATUSES (DSTA_ID, APPLICATION_ID, COMPONENT_NAME, STATUS_LEVEL, REFERENCE_ID, SOFTWARE_VERSION, TIME_STAMP, VERSION, DATE_CREATED, CREATED_BY, DAT\nE_MODIFIED, MODIFIED_BY, PPPC_ID, PPRC_ID) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [params=(long) 230503, (String) JTX, (String) input, (short) 1, (null) null, (String)\n 19-60-08, (Timestamp) 2009-02-10 17:15:05.49, (int) 1, (Timestamp) 2009-02-10 17:15:05.493, (String) weblogic, (Timestamp) 2009-02-10 17:15:05.493, (String) weblogic, (null) nul\nl, (null) null]} [code=1, state=23000]\n        at org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4238)\n        at org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4203)\n        at org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n        at org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n        at org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushAndUpdate(PreparedStatementManagerImpl.java:131)\n        at org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushInternal(PreparedStatementManagerImpl.java:89)\n        at org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flush(PreparedStatementManagerImpl.java:72)\n        at org.apache.openjpa.jdbc.kernel.OperationOrderUpdateManager.flushPrimaryRow(OperationOrderUpdateManager.java:203)\n        at org.apache.openjpa.jdbc.kernel.OperationOrderUpdateManager.flush(OperationOrderUpdateManager.java:89)\n        at org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:89)\n        at org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:72)\n        at org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:655)\n    ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"created":"2009-02-11T11:45:16.267+0000","updated":"2009-02-11T11:46:50.091+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672608","id":"12672608","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"body":"tried with the openjpa.Multithreaded  as true for oracle and didnt help","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"created":"2009-02-11T11:45:59.796+0000","updated":"2009-02-11T11:45:59.796+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12672842","id":"12672842","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi again Joe, I can now replicate the problem on both postgresql and oracle databases. Here is the error I get on oracle. Now, I will determine the cause for each. Thanks for your help.....\n\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743437M nonfatal general error> org.apache.openjpa.persistence.PersistenceException: ORA-00001: unique constraint (SYSTEM.SYS_C004001) violated\n {prepstmnt 30673895 INSERT INTO ENTITY_PERSON (id, firstName, lastName) VALUES (?, ?, ?) [params=(int) 15662, (String) First_name_1011, (String) Last_name_1011]} [code=1, state=23000]\nFailedObject: org.apache.openjpa.persistence.hightps.sequence.EntityPerson@114c8b6\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4244)\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4209)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushAndUpdate(PreparedStatementManagerImpl.java:127)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.batchOrExecuteRow(BatchingPreparedStatementManagerImpl.java:100)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushAndUpdate(BatchingPreparedStatementManagerImpl.java:84)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushInternal(PreparedStatementManagerImpl.java:93)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flush(PreparedStatementManagerImpl.java:81)\n\tat org.apache.openjpa.jdbc.kernel.ConstraintUpdateManager.flush(ConstraintUpdateManager.java:549)\n\tat org.apache.openjpa.jdbc.kernel.ConstraintUpdateManager.flush(ConstraintUpdateManager.java:106)\n\tat org.apache.openjpa.jdbc.kernel.BatchingConstraintUpdateManager.flush(BatchingConstraintUpdateManager.java:59)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:89)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:72)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:630)\n\tat org.apache.openjpa.kernel.DelegatingStoreManager.flush(DelegatingStoreManager.java:130)\n\t... 15 more\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-12T00:51:35.205+0000","updated":"2009-02-12T00:51:35.205+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12673900","id":"12673900","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Failure for DB2 below. This was to demonstrate that it is not just a Oracle and PostgreSQL problem. There seems to be a thread-safe problem in OpenJPA for all databases that support native Sequences.\n\ntestMultiThreadedLoad(org.apache.openjpa.persistence.hightps.sequence.TestSequence)  Time elapsed: 178.797 sec  <<< ERROR!\norg.apache.openjpa.persistence.hightps.sequence.TestSequence$ThreadingException: The 3 embedded errors occured in the execution of 8 iterations of 6 threads: [reflection invocation: (testMultiThreadedLoad)]\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.mttest(TestSequence.java:466)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.mttest(TestSequence.java:360)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.mttest(TestSequence.java:310)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.mttest(TestSequence.java:304)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.testMultiThreadedLoad(TestSequence.java:255)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat junit.framework.TestCase.runTest(TestCase.java:154)\n\tat junit.framework.TestCase.runBare(TestCase.java:127)\n\tat org.apache.openjpa.persistence.test.PersistenceTestCase.runBare(PersistenceTestCase.java:443)\n\tat junit.framework.TestResult$1.protect(TestResult.java:106)\n\tat junit.framework.TestResult.runProtected(TestResult.java:124)\n\tat junit.framework.TestResult.run(TestResult.java:109)\n\tat junit.framework.TestCase.run(TestCase.java:118)\n\tat org.apache.openjpa.persistence.test.PersistenceTestCase.run(PersistenceTestCase.java:173)\n\tat junit.framework.TestSuite.runTest(TestSuite.java:208)\n\tat junit.framework.TestSuite.run(TestSuite.java:203)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.maven.surefire.junit.JUnitTestSet.execute(JUnitTestSet.java:213)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:140)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:127)\n\tat org.apache.maven.surefire.Surefire.run(Surefire.java:177)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:334)\n\tat org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:980)\nNested Throwable #1: org.apache.openjpa.persistence.hightps.sequence.TestSequence$ThreadingException: thread=Thread[reflection invocation: (testMultiThreadedLoad) [5 of 6],5,main];threadNum=5;maxThreads=6;iteration=1;maxIterations=8\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:429)\nNested Throwable #1: java.lang.reflect.InvocationTargetException\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$1.run(TestSequence.java:363)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:422)\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal store error> org.apache.openjpa.persistence.RollbackException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:530)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.createManyPersonsInSeparateTransactions(TestSequence.java:134)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.testMultiThreadedLoad(TestSequence.java:261)\n\t... 6 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal general error> org.apache.openjpa.persistence.PersistenceException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.kernel.BrokerImpl.newFlushException(BrokerImpl.java:2264)\n\tat org.apache.openjpa.kernel.BrokerImpl.flush(BrokerImpl.java:2111)\n\tat org.apache.openjpa.kernel.BrokerImpl.flushSafe(BrokerImpl.java:2009)\n\tat org.apache.openjpa.kernel.BrokerImpl.beforeCompletion(BrokerImpl.java:1927)\n\tat org.apache.openjpa.kernel.LocalManagedRuntime.commit(LocalManagedRuntime.java:81)\n\tat org.apache.openjpa.kernel.BrokerImpl.commit(BrokerImpl.java:1451)\n\tat org.apache.openjpa.kernel.DelegatingBroker.commit(DelegatingBroker.java:895)\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:519)\n\t... 8 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M nonfatal general error> org.apache.openjpa.persistence.PersistenceException: One or more values in the INSERT statement, UPDATE statement, or foreign key update caused by a DELETE statement are not valid because the primary key, unique constraint or unique index identified by \"1\" constrains table \"DB2ADMIN.ENTITY_PERSON\" from having duplicate rows for those columns.\nFailedObject: prepstmnt 30057146 INSERT INTO ENTITY_PERSON (id, firstName, lastName) VALUES (?, ?, ?) [org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement]\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4244)\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4209)\n\tat org.apache.openjpa.jdbc.sql.DB2Dictionary.newStoreException(DB2Dictionary.java:507)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:193)\n\tat org.apache.openjpa.jdbc.kernel.BatchingConstraintUpdateManager.flush(BatchingConstraintUpdateManager.java:63)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:89)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:72)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:630)\n\tat org.apache.openjpa.kernel.DelegatingStoreManager.flush(DelegatingStoreManager.java:130)\n\t... 15 more\nCaused by: com.ibm.db2.jcc.b.SQLException: One or more values in the INSERT statement, UPDATE statement, or foreign key update caused by a DELETE statement are not valid because the primary key, unique constraint or unique index identified by \"1\" constrains table \"DB2ADMIN.ENTITY_PERSON\" from having duplicate rows for those columns.\n\tat com.ibm.db2.jcc.b.ce.d(ce.java:975)\n\tat com.ibm.db2.jcc.a.bd.k(bd.java:312)\n\tat com.ibm.db2.jcc.a.bd.a(bd.java:61)\n\tat com.ibm.db2.jcc.a.r.a(r.java:64)\n\tat com.ibm.db2.jcc.a.bq.c(bq.java:217)\n\tat com.ibm.db2.jcc.b.cf.C(cf.java:1109)\n\tat com.ibm.db2.jcc.b.cf.a(cf.java:1505)\n\tat com.ibm.db2.jcc.b.cf.executeUpdate(cf.java:322)\n\tat org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:101)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator$LoggingConnection$LoggingPreparedStatement.executeUpdate(LoggingConnectionDecorator.java:981)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement.executeUpdate(JDBCStoreManager.java:1501)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.executeUpdate(PreparedStatementManagerImpl.java:249)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushSingleRow(BatchingPreparedStatementManagerImpl.java:215)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:154)\n\t... 20 more\nNested Throwable #2: org.apache.openjpa.persistence.hightps.sequence.TestSequence$ThreadingException: thread=Thread[reflection invocation: (testMultiThreadedLoad) [2 of 6],5,main];threadNum=2;maxThreads=6;iteration=1;maxIterations=8\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:429)\nNested Throwable #1: java.lang.reflect.InvocationTargetException\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$1.run(TestSequence.java:363)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:422)\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal store error> org.apache.openjpa.persistence.RollbackException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:530)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.createManyPersonsAndEmployeesInSeparateTransactions(TestSequence.java:179)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.testMultiThreadedLoad(TestSequence.java:267)\n\t... 6 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal general error> org.apache.openjpa.persistence.PersistenceException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.kernel.BrokerImpl.newFlushException(BrokerImpl.java:2264)\n\tat org.apache.openjpa.kernel.BrokerImpl.flush(BrokerImpl.java:2111)\n\tat org.apache.openjpa.kernel.BrokerImpl.flushSafe(BrokerImpl.java:2009)\n\tat org.apache.openjpa.kernel.BrokerImpl.beforeCompletion(BrokerImpl.java:1927)\n\tat org.apache.openjpa.kernel.LocalManagedRuntime.commit(LocalManagedRuntime.java:81)\n\tat org.apache.openjpa.kernel.BrokerImpl.commit(BrokerImpl.java:1451)\n\tat org.apache.openjpa.kernel.DelegatingBroker.commit(DelegatingBroker.java:895)\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:519)\n\t... 8 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M nonfatal general error> org.apache.openjpa.persistence.PersistenceException: The current transaction has been rolled back because of a deadlock or timeout.  Reason code \"2\".\nFailedObject: prepstmnt 5148820 INSERT INTO ENTITY_EMPLOYEE (id, firstName, lastName, salary) VALUES (?, ?, ?, ?) [org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement]\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4244)\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4209)\n\tat org.apache.openjpa.jdbc.sql.DB2Dictionary.newStoreException(DB2Dictionary.java:507)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:193)\n\tat org.apache.openjpa.jdbc.kernel.BatchingConstraintUpdateManager.flush(BatchingConstraintUpdateManager.java:63)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:89)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:72)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:630)\n\tat org.apache.openjpa.kernel.DelegatingStoreManager.flush(DelegatingStoreManager.java:130)\n\t... 15 more\nCaused by: com.ibm.db2.jcc.b.SQLException: The current transaction has been rolled back because of a deadlock or timeout.  Reason code \"2\".\n\tat com.ibm.db2.jcc.b.ce.e(ce.java:1093)\n\tat com.ibm.db2.jcc.a.bd.s(bd.java:678)\n\tat com.ibm.db2.jcc.a.bd.k(bd.java:335)\n\tat com.ibm.db2.jcc.a.bd.a(bd.java:61)\n\tat com.ibm.db2.jcc.a.r.a(r.java:64)\n\tat com.ibm.db2.jcc.a.bq.c(bq.java:217)\n\tat com.ibm.db2.jcc.b.cf.C(cf.java:1109)\n\tat com.ibm.db2.jcc.b.cf.a(cf.java:1505)\n\tat com.ibm.db2.jcc.b.cf.executeUpdate(cf.java:322)\n\tat org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:101)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator$LoggingConnection$LoggingPreparedStatement.executeUpdate(LoggingConnectionDecorator.java:981)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement.executeUpdate(JDBCStoreManager.java:1501)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.executeUpdate(PreparedStatementManagerImpl.java:249)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushSingleRow(BatchingPreparedStatementManagerImpl.java:215)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:154)\n\t... 20 more\nNested Throwable #3: org.apache.openjpa.persistence.hightps.sequence.TestSequence$ThreadingException: thread=Thread[reflection invocation: (testMultiThreadedLoad) [4 of 6],5,main];threadNum=4;maxThreads=6;iteration=1;maxIterations=8\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:429)\nNested Throwable #1: java.lang.reflect.InvocationTargetException\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$1.run(TestSequence.java:363)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence$2.run(TestSequence.java:422)\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal store error> org.apache.openjpa.persistence.RollbackException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:530)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.createManyPersonsAndEmployeesInSeparateTransactions(TestSequence.java:179)\n\tat org.apache.openjpa.persistence.hightps.sequence.TestSequence.testMultiThreadedLoad(TestSequence.java:267)\n\t... 6 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M fatal general error> org.apache.openjpa.persistence.PersistenceException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.kernel.BrokerImpl.newFlushException(BrokerImpl.java:2264)\n\tat org.apache.openjpa.kernel.BrokerImpl.flush(BrokerImpl.java:2111)\n\tat org.apache.openjpa.kernel.BrokerImpl.flushSafe(BrokerImpl.java:2009)\n\tat org.apache.openjpa.kernel.BrokerImpl.beforeCompletion(BrokerImpl.java:1927)\n\tat org.apache.openjpa.kernel.LocalManagedRuntime.commit(LocalManagedRuntime.java:81)\n\tat org.apache.openjpa.kernel.BrokerImpl.commit(BrokerImpl.java:1451)\n\tat org.apache.openjpa.kernel.DelegatingBroker.commit(DelegatingBroker.java:895)\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:519)\n\t... 8 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:743836M nonfatal general error> org.apache.openjpa.persistence.PersistenceException: The current transaction has been rolled back because of a deadlock or timeout.  Reason code \"2\".\nFailedObject: prepstmnt 6639365 INSERT INTO ENTITY_EMPLOYEE (id, firstName, lastName, salary) VALUES (?, ?, ?, ?) [org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement]\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4244)\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4209)\n\tat org.apache.openjpa.jdbc.sql.DB2Dictionary.newStoreException(DB2Dictionary.java:507)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:193)\n\tat org.apache.openjpa.jdbc.kernel.BatchingConstraintUpdateManager.flush(BatchingConstraintUpdateManager.java:63)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:89)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:72)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:630)\n\tat org.apache.openjpa.kernel.DelegatingStoreManager.flush(DelegatingStoreManager.java:130)\n\t... 15 more\nCaused by: com.ibm.db2.jcc.b.SQLException: The current transaction has been rolled back because of a deadlock or timeout.  Reason code \"2\".\n\tat com.ibm.db2.jcc.b.ce.e(ce.java:1093)\n\tat com.ibm.db2.jcc.a.bd.s(bd.java:678)\n\tat com.ibm.db2.jcc.a.bd.k(bd.java:335)\n\tat com.ibm.db2.jcc.a.bd.a(bd.java:61)\n\tat com.ibm.db2.jcc.a.r.a(r.java:64)\n\tat com.ibm.db2.jcc.a.bq.c(bq.java:217)\n\tat com.ibm.db2.jcc.b.cf.C(cf.java:1109)\n\tat com.ibm.db2.jcc.b.cf.a(cf.java:1505)\n\tat com.ibm.db2.jcc.b.cf.executeUpdate(cf.java:322)\n\tat org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:101)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator$LoggingConnection$LoggingPreparedStatement.executeUpdate(LoggingConnectionDecorator.java:981)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:269)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement.executeUpdate(JDBCStoreManager.java:1501)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.executeUpdate(PreparedStatementManagerImpl.java:249)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushSingleRow(BatchingPreparedStatementManagerImpl.java:215)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushBatch(BatchingPreparedStatementManagerImpl.java:154)\n\t... 20 more","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-16T15:17:53.567+0000","updated":"2009-02-16T15:17:53.567+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12674088","id":"12674088","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"The attached patch provides a testcase to demonstrate the failure on both PostgreSQL and Oracle, and likewise provides a fix for both PostgreSQL and Oracle. I would suggest that it be applied to both Trunk and the 1.2.x branch. Thanks","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-17T02:40:48.721+0000","updated":"2009-02-17T02:40:48.721+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12676381","id":"12676381","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"body":"Hi Tim, thanks for the patch I can guarantee to anyone who is using OpenJPA with a larger configuration (4 solaris machines with 24 core each) and oracle sequences that this patch is a MUST.\nAfter testing with the patch our problems with duplicate pk from sequences have disappeared. Also the patch doesnt seem to have a major impact on performance.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"created":"2009-02-24T19:43:47.796+0000","updated":"2009-02-24T19:43:47.796+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12677039","id":"12677039","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"HI Joe, that's wonderful new !! Thanks much for verifying that it does it fact work for your configuration.....","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-26T15:27:03.970+0000","updated":"2009-02-26T15:27:03.970+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12677634","id":"12677634","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Closing since we now have verification that it fixes the problem.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-02-28T03:04:25.629+0000","updated":"2009-02-28T03:04:25.629+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12693863","id":"12693863","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"Although the patch applied (synchronization on PreparedStatement within NativeJDBCSeq.getSequence) does a good job on reducing the likelihood of crash, I am afraid we are still not free from multi-threading issue. On my one-core laptop I am not able to reproduce this issue (even when I remove the patch) but I find the problem still exists in the AbstractJDBCSeq.next method. I am able to reproduce the exception when I insert Thread.yield():\n\n    public Object next(StoreContext ctx, ClassMetaData meta) {\n        JDBCStore store = getStore(ctx);\n        try {\n            current = nextInternal(store, (ClassMapping) meta);\n            Thread.yield();\n            return current;\n        } catch (OpenJPAException ke) {\n            throw ke;\n        } catch (SQLException se) {\n            throw SQLExceptions.getStore(se, store.getDBDictionary());\n        } catch (Exception e) {\n            throw new StoreException(e);\n        }\n    }\n\nThis yield() simulates context switching in an real application. The context switch will actually seldom occur here but is possible to my knowledge. Also, I am able to reproduce the exception by adding some Thread.sleep calls instead of yield() but this is harder to reproduce.\n\nMy suggestion is to remove the synchronization from NativeJDBCSeq.getSequence and modify AbstractJDBCSeq.next to something like the following:\n\n    public Object next(StoreContext ctx, ClassMetaData meta) {\n        JDBCStore store = getStore(ctx);\n        try {\n                Object currentLocal = nextInternal(store, (ClassMapping) meta);\n                current = currentLocal;\n                return currentLocal;\n        } catch (OpenJPAException ke) {\n            throw ke;\n        } catch (SQLException se) {\n            throw SQLExceptions.getStore(se, store.getDBDictionary());\n        } catch (Exception e) {\n            throw new StoreException(e);\n        }\n    }\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-03-30T17:49:59.083+0000","updated":"2009-03-30T17:49:59.083+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12703025","id":"12703025","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"The attached patch implements my suggestion. I think this is a better solution than we currently have because:\n- the multi-threading problem will be completely avoided to my knowledge\n- by removing synchronization from PreparedStatement execution we gain a bit of scalability.\n\nI am going to commit it unless somebody has objection.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-04-27T08:21:33.478+0000","updated":"2009-04-27T08:21:33.478+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12708537","id":"12708537","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"I have applied the patch to trunk and 1.3.x branch.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-12T18:04:07.781+0000","updated":"2009-05-12T18:04:07.781+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12709197","id":"12709197","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Milosz, It's a little unclear to me how these changes will prevent duplicate sequence values. And actually, I can easily create duplicates with these changes in trunk using the postgreSQL database (by altering the number of entities created in the TestSequence testcase). See the error/exception messages below and look for \"duplicate key value violates unique constraint\", which is indicative of duplicate sequences getting created. This appears to be a regression as these exceptions were eliminated using the previous patch.....\n\n-------------------------------------------------------\n T E S T S\n-------------------------------------------------------\nRunning org.apache.openjpa.persistence.sequence.TestSequence\n359  test  INFO   [main] openjpa.jdbc.JDBC - Using dictionary class \"org.apache.openjpa.jdbc.sql.PostgresDictionary\" (PostgreSQL 8.3.6 ,PostgreSQL Native Driver PostgreSQL 8.3 JDBC3 with SSL (build 603)).\n1406  test  INFO   [reflection invocation: (testMultiThreadedNativeSequences) [6 of 6]] openjpa.Runtime - Starting OpenJPA 2.0.0-SNAPSHOT\nTests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 361.859 sec <<< FAILURE!\ntestMultiThreadedNativeSequences(org.apache.openjpa.persistence.sequence.TestSequence)  Time elapsed: 361.765 sec  <<< ERROR!\norg.apache.openjpa.persistence.sequence.TestSequence$ThreadingException: The 1 embedded errors occured in the execution of 8 iterations of 6 threads: [reflection invocation: (testMultiThreadedNativeSequences)]\n\tat org.apache.openjpa.persistence.sequence.TestSequence.mttest(TestSequence.java:440)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.mttest(TestSequence.java:334)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.mttest(TestSequence.java:281)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.mttest(TestSequence.java:275)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.testMultiThreadedNativeSequences(TestSequence.java:66)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat junit.framework.TestCase.runTest(TestCase.java:154)\n\tat junit.framework.TestCase.runBare(TestCase.java:127)\n\tat org.apache.openjpa.persistence.test.PersistenceTestCase.runBare(PersistenceTestCase.java:465)\n\tat junit.framework.TestResult$1.protect(TestResult.java:106)\n\tat junit.framework.TestResult.runProtected(TestResult.java:124)\n\tat junit.framework.TestResult.run(TestResult.java:109)\n\tat junit.framework.TestCase.run(TestCase.java:118)\n\tat org.apache.openjpa.persistence.test.PersistenceTestCase.run(PersistenceTestCase.java:181)\n\tat junit.framework.TestSuite.runTest(TestSuite.java:208)\n\tat junit.framework.TestSuite.run(TestSuite.java:203)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.maven.surefire.junit.JUnitTestSet.execute(JUnitTestSet.java:213)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:140)\n\tat org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:127)\n\tat org.apache.maven.surefire.Surefire.run(Surefire.java:177)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:345)\n\tat org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:1009)\nNested Throwable #1: org.apache.openjpa.persistence.sequence.TestSequence$ThreadingException: thread=Thread[reflection invocation: (testMultiThreadedNativeSequences) [1 of 6],5,main];threadNum=1;maxThreads=6;iteration=2;maxIterations=8\n\tat org.apache.openjpa.persistence.sequence.TestSequence$2.run(TestSequence.java:405)\nNested Throwable #1: java.lang.reflect.InvocationTargetException\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:585)\n\tat org.apache.openjpa.persistence.sequence.TestSequence$1.run(TestSequence.java:337)\n\tat org.apache.openjpa.persistence.sequence.TestSequence$2.run(TestSequence.java:398)\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:774393M fatal store error> org.apache.openjpa.persistence.RollbackException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:559)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.createManyPersonsAndEmployeesInSeparateTransactions(TestSequence.java:191)\n\tat org.apache.openjpa.persistence.sequence.TestSequence.testMultiThreadedNativeSequences(TestSequence.java:78)\n\t... 6 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:774393M fatal general error> org.apache.openjpa.persistence.PersistenceException: The transaction has been rolled back.  See the nested exceptions for details on the errors that occurred.\n\tat org.apache.openjpa.kernel.BrokerImpl.newFlushException(BrokerImpl.java:2266)\n\tat org.apache.openjpa.kernel.BrokerImpl.flush(BrokerImpl.java:2113)\n\tat org.apache.openjpa.kernel.BrokerImpl.flushSafe(BrokerImpl.java:2011)\n\tat org.apache.openjpa.kernel.BrokerImpl.beforeCompletion(BrokerImpl.java:1929)\n\tat org.apache.openjpa.kernel.LocalManagedRuntime.commit(LocalManagedRuntime.java:81)\n\tat org.apache.openjpa.kernel.BrokerImpl.commit(BrokerImpl.java:1453)\n\tat org.apache.openjpa.kernel.DelegatingBroker.commit(DelegatingBroker.java:895)\n\tat org.apache.openjpa.persistence.EntityManagerImpl.commit(EntityManagerImpl.java:548)\n\t... 8 more\nCaused by: <openjpa-2.0.0-SNAPSHOT-r422266:774393M fatal general error> org.apache.openjpa.persistence.PersistenceException: ERROR: duplicate key value violates unique constraint \"entity_employee_pkey\" {prepstmnt 24724157 INSERT INTO ENTITY_EMPLOYEE (id, firstName, lastName, salary) VALUES (?, ?, ?, ?) [params=(int) 256631, (String) 4_First_name_6912, (String) 4_Last_name_6912, (float) 6912.0]} [code=0, state=23505]\nFailedObject: org.apache.openjpa.persistence.sequence.EntityEmployee@ff5a2b\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.narrow(DBDictionary.java:4383)\n\tat org.apache.openjpa.jdbc.sql.DBDictionary.newStoreException(DBDictionary.java:4342)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:102)\n\tat org.apache.openjpa.jdbc.sql.SQLExceptions.getStore(SQLExceptions.java:72)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushAndUpdate(PreparedStatementManagerImpl.java:141)\n\tat org.apache.openjpa.jdbc.kernel.BatchingPreparedStatementManagerImpl.flushAndUpdate(BatchingPreparedStatementManagerImpl.java:80)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushInternal(PreparedStatementManagerImpl.java:97)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flush(PreparedStatementManagerImpl.java:85)\n\tat org.apache.openjpa.jdbc.kernel.ConstraintUpdateManager.flush(ConstraintUpdateManager.java:550)\n\tat org.apache.openjpa.jdbc.kernel.ConstraintUpdateManager.flush(ConstraintUpdateManager.java:106)\n\tat org.apache.openjpa.jdbc.kernel.BatchingConstraintUpdateManager.flush(BatchingConstraintUpdateManager.java:59)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:103)\n\tat org.apache.openjpa.jdbc.kernel.AbstractUpdateManager.flush(AbstractUpdateManager.java:76)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager.flush(JDBCStoreManager.java:672)\n\tat org.apache.openjpa.kernel.DelegatingStoreManager.flush(DelegatingStoreManager.java:130)\n\t... 15 more\nCaused by: org.apache.openjpa.lib.jdbc.ReportingSQLException: ERROR: duplicate key value violates unique constraint \"entity_employee_pkey\" {prepstmnt 24724157 INSERT INTO ENTITY_EMPLOYEE (id, firstName, lastName, salary) VALUES (?, ?, ?, ?) [params=(int) 256631, (String) 4_First_name_6912, (String) 4_Last_name_6912, (float) 6912.0]} [code=0, state=23505]\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator.wrap(LoggingConnectionDecorator.java:236)\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator.access$700(LoggingConnectionDecorator.java:69)\n\tat org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator$LoggingConnection$LoggingPreparedStatement.executeUpdate(LoggingConnectionDecorator.java:1080)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:286)\n\tat org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:286)\n\tat org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement.executeUpdate(JDBCStoreManager.java:1579)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.executeUpdate(PreparedStatementManagerImpl.java:246)\n\tat org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushAndUpdate(PreparedStatementManagerImpl.java:116)\n\t... 25 more\n\n\nResults :\n\nTests in error: \n  testMultiThreadedNativeSequences(org.apache.openjpa.persistence.sequence.TestSequence)\n\nTests run: 1, Failures: 0, Errors: 1, Skipped: 0\n\n[INFO] ------------------------------------------------------------------------\n[ERROR] BUILD FAILURE\n[INFO] ------------------------------------------------------------------------\n[INFO] There are test failures.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-13T22:56:34.987+0000","updated":"2009-05-13T22:56:34.987+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12709419","id":"12709419","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kwsutter","name":"kwsutter","key":"kwsutter","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kevin W. Sutter","active":true,"timeZone":"America/Chicago"},"body":"Based on Tim's comments and test case failures, it doesn't look like this JIRA is completely resolved yet.  Milosz, please work with Tim to come to a common resolution.  Thanks.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kwsutter","name":"kwsutter","key":"kwsutter","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kevin W. Sutter","active":true,"timeZone":"America/Chicago"},"created":"2009-05-14T14:48:22.685+0000","updated":"2009-05-14T14:48:22.685+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12709525","id":"12709525","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"Hi Tim, do you run the testcase on a multi-core CPU? What exactly is your change to the testcase?\n\nI am looking into this and will let know about my findings next week. If I can't help, I will roll my patch back or merge the both patches.\n\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-14T18:47:08.939+0000","updated":"2009-05-14T18:47:08.939+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12709875","id":"12709875","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Milosz, what I did was change the number of Entities, and number of threads to try to emulate Joe's environment. Try these settings at these lines below. Also, make sure you're using a database that supports native sequences (i.e., nextSequenceQuery in the dictionary -- Derby for example, does not).\n\n00042    private static final int NUMBER_ENTITIES = 30000;\n00264    int iterations = 10;\n00265    int threads = 12;\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-15T15:22:47.087+0000","updated":"2009-05-15T15:22:47.087+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12711689","id":"12711689","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"Hi Tim, is it possible that you make the following tests?\n\n1. Apply the volatile.patch, compile and run the test and see whether it changes anything.\n2. If the above did not help, revert volatile.patch and try to reproduce the error on DB2 or Oracle, so we are sure it is not PostgreSQL fault.\n\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-05-21T16:58:25.039+0000","updated":"2009-05-21T16:58:25.039+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12722759","id":"12722759","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"Maybe some volounteer could try reproducing this and if reproduced, try with volatile.patch. If not, I will restore the synchronization around PreparedStatement since I am not able to reproduce the issue with my system (Pentium M).\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-06-22T18:52:16.292+0000","updated":"2009-06-22T18:52:16.292+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12728844","id":"12728844","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Milosz, I'll apply your volatile patch later today and let you know what  happens. Note though that the failure can I can reproduce the problem on DB2, Oracle, and PostgreSQL -- it's certainly not isolated to PostgreSQL. Talk later....","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-07-08T18:37:22.997+0000","updated":"2009-07-08T18:37:22.997+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12730114","id":"12730114","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"body":"Hi Milosz, here is what I've done over the past couple days. I have two machines -- Windows XP, Windows 2003 -- each with MySQL (8.3 on one machine, 8.4 on the other machine), DB2 (8.2 and 9.5), and Oracle (10g on both) installed on them. To reproduce the original problem I use 25000 entities, 6 iterations, and 8 threads in the TestSequence test program. With these settings I can easily reproduce the failure on both machines on all three databases. Here are the results of the various patches attached to this JIRA. Note that I raised the number entities to 40000 to stress each patch (but still used 6 iterations and 8 threads):\n\n1. Patch OPENJPA-466 (attached by you with the change to AbstractJDBCSeq.java): Failures on all three databases on both machines. Note that the failures manifest themselves differently on each database. For example the failures on PostgreSQL show up as \n\nERROR: duplicate key value violates unique constraint \n\non Oracle as: ORA-00001: unique constraint (SYSTEM.SYS_C008294) violated\n\nand finally on DB2 as:  org.apache.openjpa.persistence.EntityExistsException: DB2 SQL error: SQLCODE: -803, SQLSTATE: 23505\n\n2. Patch volite.patch only (i.e., no other patches used): Failures on all three databases on both machines\n\n3. Patch OPENJPA-466-SYNCHRONIZED.patch only (which is a modified version of my original patch but is what was originally committed to trunk): Success on all three databases on both machines.\n\nSo here are my recommendations:\n\nThe OPENJPA-466-SYNCHRONIZED patch works in all scenarios that I've tested it in. More importantly, it fixed the original problem reported by the user, and has no adverse impact on performance. Please note as well that that user environment was an extremely large, high-volume, multi-threaded, multi-core installation. Not meaning to be overly brusque -- but it simply works in every scenario that it's been used it. Until or unless someone (other than me as I don't intend to spend any more time on this JIRA) can provide a scenario where it does not work it should be committed to trunk and retrofitted to all previous versions of OpenJPA, and any previously-committed patches should be reverted. Thanks much\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mcconne","name":"mcconne","key":"mcconne","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Tim McConnell","active":true,"timeZone":"Etc/UTC"},"created":"2009-07-12T16:05:10.359+0000","updated":"2009-07-12T16:05:10.359+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12730321","id":"12730321","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"body":"Tim, not sure why you marked this as Resolved when your latest patch hasn't been committed yet.....","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"created":"2009-07-13T12:45:59.375+0000","updated":"2009-07-13T12:45:59.375+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12730370","id":"12730370","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikedd","name":"mikedd","key":"mikedd","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Michael Dick","active":true,"timeZone":"America/Chicago"},"body":"I think we're up to date in trunk and 1.3.x now. Milosz, Frank  or Tim, if there are additional changes that need to be made please re-open the issue.\n\nThanks for the patches and the time you spend debugging this problem Tim. ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikedd","name":"mikedd","key":"mikedd","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Michael Dick","active":true,"timeZone":"America/Chicago"},"created":"2009-07-13T14:58:44.892+0000","updated":"2009-07-13T14:58:44.892+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12734743","id":"12734743","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"The volatile.patch was meant to be applied to what was on the trunk without any reverts so it is not a surprise that \"2. Patch volite.patch only (i.e., no other patches used): Failures on all three databases on both machines\".\n\nI agree with applying the latest patch and resolving this. This is better that what we had before and it seems there is not enough interest in testing the patches.\n\nI do state however we just hid a multi-threading bug that can bite us later.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-07-23T19:25:26.114+0000","updated":"2009-07-23T19:25:26.114+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12747077","id":"12747077","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bjreed","name":"bjreed","key":"bjreed","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Barrett Jay Reed","active":true,"timeZone":"Etc/UTC"},"body":"Attached 2 patches to port necessary code back to 1.0.x and 1.2.x.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bjreed","name":"bjreed","key":"bjreed","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Barrett Jay Reed","active":true,"timeZone":"Etc/UTC"},"created":"2009-08-24T21:08:48.958+0000","updated":"2009-08-24T21:08:48.958+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12750419","id":"12750419","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"body":"add missing Fix versions of 1..0.4 and 1.2.2","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=drwoods","name":"drwoods","key":"drwoods","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=drwoods&avatarId=20433","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=drwoods&avatarId=20433","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=drwoods&avatarId=20433","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=drwoods&avatarId=20433"},"displayName":"Donald Woods","active":true,"timeZone":"America/New_York"},"created":"2009-09-02T12:37:47.677+0000","updated":"2009-09-02T12:37:47.677+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12759885","id":"12759885","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"body":"Today came back to this issue get the patch and was surprised that after the first fix there was more discussion. Just wanted to say from my view the original patch from Tim did the job perfectly.\n\nAs Tim stated I can tell you that the original patch was def needed and helped solve a major issue with an application running on 4 T2000 Solaris machines (each with 64 cores, yes 256 cores !). This J2EE app has processed 10 millions of business transactions with throughput in peak of 100,000s business transactions (equals to 250,000 JTA and over a million SQL statements) per hour without a SINGLE duplicate primary key error related to sequences.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=joepullen","name":"joepullen","key":"joepullen","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=joepullen&avatarId=22130","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=joepullen&avatarId=22130","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=joepullen&avatarId=22130","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=joepullen&avatarId=22130"},"displayName":"Joe Pullen","active":true,"timeZone":"Europe/Zurich"},"created":"2009-09-26T11:55:05.753+0000","updated":"2009-09-26T11:55:05.753+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12767388","id":"12767388","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"The issue of sequence value creation under multithreaded environment is still relevant. \nI ran a very simple tests that do the following \n1. A entity with @GeneratedValue primary key\n2. 4 threads that persist() entities in separate persistence contexts\n\nThe test breaks with OPENJPA_SEQUENCE_TABLE being populated with duplicate key -- looks like a concurrency issue.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-10-19T17:12:02.060+0000","updated":"2009-10-19T17:12:02.060+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12767389","id":"12767389","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"The test was run onm MYSQL 5.3.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-10-19T17:12:48.842+0000","updated":"2009-10-19T17:12:48.842+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12767390","id":"12767390","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"body":"A test and a entity to demonstrate the concurrency issue in sequence value generation.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ppoddar%40apache.org","name":"ppoddar@apache.org","key":"ppoddar@apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Pinaki Poddar","active":true,"timeZone":"America/Los_Angeles"},"created":"2009-10-19T17:15:24.450+0000","updated":"2009-10-19T17:15:24.450+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12770149","id":"12770149","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"I confirm the issue although I had to modify the test to call em.persist in a loop and be a bit patient.\n\nWe are probably failing during OPENJPA_SEQUENCE_TABLE initialization, not during generating numbers from it.\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-10-26T19:18:22.348+0000","updated":"2009-10-26T19:18:22.348+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12384540/comment/12772287","id":"12772287","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"body":"I will create a new issue for the sequence table problem as it is loosely coupled with the original issue. I guess the sequence table failure is the known problem where multiple threads access an uninitailized emf, it has been reported on the users mailing list.\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=milosz","name":"milosz","key":"milosz","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Milosz Tylenda","active":true,"timeZone":"Etc/UTC"},"created":"2009-10-31T21:13:01.342+0000","updated":"2009-10-31T21:13:01.342+0000"}],"maxResults":37,"total":37,"startAt":0},"customfield_12311820":"0|i0yvk7:","customfield_12314139":null}}

